lxpanel-plugin/kano_updater.c

```diff
@@ -380,71 +382,79 @@ void install_clicked(GtkWidget *widget, gpointer data)
     launch_cmd(SOUND_CMD, NULL);
 }
 
-void check_for_updates_clicked(GtkWidget *widget,
-			       kano_updater_plugin_t *plugin_data)
+void check_for_updates_clicked(GtkWidget *widget, kano_updater_plugin_t *plugin_data)
 {
-	launch_cmd(CHECK_FOR_UPDATES_CMD,  "kano-updater");
+    // launch_cmd can't be used because we need to wait for 
+    // the command to finish
+    kdesk_hourglass_start("kano-dialog");
+    system(CHECK_FOR_UPDATES_CMD);
```

Comment:
My only concern here is exactly the reason that `system` was used rather than `launch_cmd` - because it blocks. Would this freeze the UI? If so, could this be run in a thread instead?